### ISSUE TITLE
Extension makes vscode terminals unusable as soon as a contract file is opened

### ISSUE THREAD

**Issue by j-mars** [juanfranblanco/vscode-solidity#166]
In version 0.0.7.1 for WSL, as soon as I open a contract, the embedded compiler starts (in the output window) and the other terminals I have opened become unusable/extremely slow to type.

This behaviour is not normal and is forcing me to restart VSCode everytime but I am considering to uninstall the extension.

What might be causing this?

**Comment by juanfranblanco** [comment]
@j-mars On 71 I initialise the compiler when the extension loaded to optimise start up of the compiler. It did perform ok on a normal environment, although I did not test it on WSL so I have removed it for now and just load it on the first compilation.

**Comment by j-mars** [comment]
Thank you for the answer. 

I have updated to 72 and I believe the issue still remains. It is not just the terminals which are affect, it is the whole behaviour of VSCode.
For instance, trying to open a contract file (in comparison mode, after changes), notice the loading bar and how the contents do not display (only after 5-10 seconds):

![image](https://user-images.githubusercontent.com/45543880/78049474-02956580-7373-11ea-9470-176ce6311897.png)

**Comment by juanfranblanco** [comment]
@j-mars thanks for this, so you are using WSL remote correct?

**Comment by j-mars** [comment]
@juanfranblanco yes correct.

**Comment by juanfranblanco** [comment]
@j-mars I tried to reproduce it in WSL but I have not had any issues. The only thing this can be related could be to an upgrade to the prettier solidity plugin which it makes it slower or notification on changes on settings that will trigger to reload the compiler. I don't think is the latter as surely you are not changing that, so can you try disabling the prettier formatting for WSL or User here: 

![image](https://user-images.githubusercontent.com/562371/78140013-4646a880-7421-11ea-8dac-ad676c532644.png)

Let me know if that solves it.

**Comment by j-mars** [comment]
@juanfranblanco I changed back to 72, tried with that setting disabled, the same thing happens: there is a large delay when typing in the terminal, sometimes it gets stuck, sometimes the contracts do not even load up.

Is there anything else I can do ?

**Comment by juanfranblanco** [comment]
@j-mars what compiler settings do you use?

**Comment by j-mars** [comment]
> @j-mars what compiler settings do you use?

I wanted to use solc version downloaded via truffle which is installed globally via yarn, but I am unsure your extension is picking it up. Not sure if playing around with any other setting helps.

![image](https://user-images.githubusercontent.com/45543880/78230504-53b76d80-74c9-11ea-9cba-3086de714ea5.png)
![image](https://user-images.githubusercontent.com/45543880/78230558-66ca3d80-74c9-11ea-8f28-488dcefaf426.png)

**Comment by juanfranblanco** [comment]
Ahh, you will need to specify the solidity version the extension cannot pick that.  Check these screenshots to set the compiler version https://github.com/juanfranblanco/vscode-solidity/issues/171#issuecomment-609362266

**Comment by j-mars** [comment]
@juanfranblanco I tried that and tested for about 5 minutes.

![image](https://user-images.githubusercontent.com/45543880/78927487-e7270a80-7a96-11ea-8cad-e2f0390d6609.png)

The stutters/slowdowns are still present, mainly when typing and opening/splitting terminals in vscode.

**Comment by juanfranblanco** [comment]
Hmm the only thing that I can think is to disable the server side doing the validation of the document:
![image](https://user-images.githubusercontent.com/562371/78928231-2b66da80-7a98-11ea-928b-33f294cac1d2.png)
or 
![image](https://user-images.githubusercontent.com/562371/78928296-40436e00-7a98-11ea-9829-bd200c65b76c.png)

**Comment by j-mars** [comment]
Did both, together and separate, Somehow there is a placebo effect that things are good in the beginning, however the slowdowns start soon. For instance, after I split to a third terminal on the same window, it got completely stuck (not possible to type - frozen):

![image](https://user-images.githubusercontent.com/45543880/78980478-e2ee0200-7b15-11ea-856e-1b95ef8e6e6d.png)

Had to fully close vscode and start again.

Issue is aggravated if vscode is opened with a contract already opened. If a contract (any) is opened only after vscode starts, issue seems to start to happen only after then (and remains for the rest of the vscode session).

I think I have tried everything, I will continue to use version 0.70, although I highly doubt I am the only one using WSL in W10 having this issue.

**Comment by juanfranblanco** [comment]
@j-mars Hmm I guess if the problem is with the WSL extension and opening panels that triggers a settings change kind of like a new workspace open. Ill try to see if removing that helps on the next release. This should not be an issue obviously, but surely this is strange.

Thank you very much, btw.

**Comment by j-mars** [comment]
@juanfranblanco  Yes, the problem seems exacerbated when I start to open multiple panels (which I need!)

Thanks for your help as well

**Comment by juanfranblanco** [comment]
@j-mars  can you check the latest and see if it helps https://github.com/juanfranblanco/vscode-solidity/blob/master/solidity-0.0.73.vsix.

**Comment by j-mars** [comment]
> @j-mars can you check the latest and see if it helps https://github.com/juanfranblanco/vscode-solidity/blob/master/solidity-0.0.73.vsix.

Tested for about 20 mins, both with a contract file opened on Vscode start and not.

If a contract file is **not open on start**, some optimization is quite noticeable now, a lot less lag, especially when only 1 or two terminals are open/running code. **However**, f I open 3 or 4 terminals with running code, problems start and some terminals start getting stuck, but I think we are on to something as it is overall better than version 72.

If I start Vscode with a contract file open already open in the files tab, the program freezes for a long time and the first terminal takes a long time to start. Even after the problems are listed (about 15 seconds in) there is a huge noticeable lag with just one terminal open, so the program is unusable.

Hope it helps with some directions. I wonder why having a contract open when vscode starts affects so much.
